src/netex_france/stops.rs

```diff
@@ -0,0 +1,141 @@
+}
+
+// Publicly exposed methods
+impl<'a> StopPointExporter<'a> {
+    pub fn new(
+        model: &'a Model,
+        participant_ref: &'a str,
+        stop_provider_code: Option<&'a String>,
+    ) -> Self {
+        StopPointExporter {
+            model,
+            participant_ref,
+            stop_provider_code,
+        }
+    }
+    pub fn export(&self) -> Result<Vec<Element>> {
+        self.model
+            .stop_points
+            .values()
+            .map(|stop_point| self.export_stop_point(stop_point))
+            .collect()
+    }
+}
+
+// Internal methods
+impl<'a> StopPointExporter<'a> {
+    fn export_stop_point(&self, stop_point: &'a StopPoint) -> Result<Element> {
+        let element_builder = Element::builder("Quay")
+            .attr("id", self.generate_id(stop_point))
+            .attr("version", "any");
+        let element_builder = element_builder.append(self.generate_name(stop_point));
+        let element_builder = if let Some(public_code) = self.generate_public_code(stop_point) {
+            element_builder.append(public_code)
+        } else {
+            element_builder
+        };
+        let element_builder = element_builder.append(self.generate_site_ref(stop_point));
+        let element_builder = element_builder.append(self.generate_centroid(stop_point)?);
+        let element_builder = element_builder.append(self.generate_transport_mode(stop_point));
+        let element_builder = if let Some(tariff_zones) = self.generate_tariff_zones(stop_point) {
+            element_builder.append(tariff_zones)
+        } else {
+            element_builder
+        };
+        Ok(element_builder.build())
+    }
+
+    fn generate_id(&self, stop_point: &'a StopPoint) -> String {
+        let id = stop_point.id.replace(':', "_");
+        let provider_code = self
+            .stop_provider_code
+            .cloned()
+            .unwrap_or_else(|| String::from("LOC"));
+        // TODO: Find INSEE code from geolocation of the `stop_point`
+        let insee = "XXXXX";
+        format!("FR:{}:ZE:{}:{}", insee, id, provider_code)
+    }
+
+    fn generate_name(&self, stop_point: &'a StopPoint) -> Element {
+        Element::builder("Name")
+            .append(Node::Text(stop_point.name.to_owned()))
+            .build()
+    }
+
+    fn generate_public_code(&self, stop_point: &'a StopPoint) -> Option<Element> {
+        stop_point.code.as_ref().map(|code| {
+            Element::builder("PublicCode")
+                .append(Node::Text(code.to_owned()))
+                .build()
+        })
+    }
+
+    fn generate_centroid(&self, stop_point: &'a StopPoint) -> Result<Element> {
+        // FIXME: String 'EPSG:4326' is failing at runtime (string below is equivalent but works)
+        let from = "+proj=longlat +datum=WGS84 +no_defs"; // See https://epsg.io/4326
+        let to = "EPSG:2154";
+        let converter = Proj::new_known_crs(from, to, None).ok_or_else(|| {
```

Comment:
Will we have a performance issue if we instantiate proj for each stop point?

Reply:
Good point, I should be able to create a field of `StopPointExporter` and initialize it at the construction... I'll rework that.

Reply:
Done in 60398174f0e609ddee1a4ae80833960800f483b5.